providers/implementations/ciphers/cipher_aes_cts_fips.c

```diff
@@ -100,6 +101,9 @@ static size_t cts128_cs1_encrypt(PROV_CIPHER_CTX *ctx, const unsigned char *in,
     aligned_16bytes tmp_in;
     size_t residue;
 
+    if (!provider_is_running())
```

Comment:
I think this is too low level..
The lower we do it the easier it will be to miss something.